### ISSUE TITLE
"Invalid cursor" error when using resources instead of triggers

### ISSUE THREAD

**Issue by codebycaleb** [zapier/zapier-platform#24]
## What would you like to change?

When calling `z.cursor.get` or `z.cursor.set` in a trigger, the `appTester` does not raise an error.

However, copying the same code to a `resource`'s `list.operation.perform` and running a test against that resource will raise an error:

```
Invalid cursor operation. `z.cursor` can only be used in `perform` methods of polling triggers and when `canPaginate` is `true`.
```

Investigation seems to point to https://github.com/zapier/zapier-platform-core/blob/48ddbd4a355723bd04143ef64da9319c1d4e1ebf/src/tools/create-app-tester.js#L9-L21 being the area that needs adjusted (it checks for triggers but not resources).

## Your Environment

* CLI Version (`zapier -v`): 
```
zapier-platform-cli/8.1.0
zapier-platform-core/8.1.0
node/v10.5.0
```
* App id: 20367

**Comment by codajonathan** [comment]
Wanted to check in on the status of this. Is this in fact a bug?

We're having the same issue. Our application is moving from offset-based pagination to cursor-based pagination. Our resources are set to `canPaginate: true`. It was easy to implement `perform` to get and set the cursor as needed.

However, when using `createAppTester`, we are now getting 

```
Invalid cursor operation. `z.cursor` can only be used in `perform` methods of polling triggers and when `canPaginate` is `true`.
```

It seems like this is supported by the app tester is incorrectly raising an issue here. Is that the case? Is there anything we can do to work around this in the meantime?

**Comment by xavdid** [comment]
I think it's just an issue with `createAppTester` and resources w/ paginating triggers. 

If you don't mind, could you post your `appTester(...` line for the trigger that doesn't work? That'll help me narrow this down.

**Comment by codajonathan** [comment]
Thanks @xavdid !

The test itself is basically:

```
const bundle = {
  ...fake bundle stuff
};

...

bundle.meta.page = 2;
const results = await appTester(App.resources.column.list.operation.perform, bundle);
```

And the definition of the resource in question is:

```
async function performListColumns(z: ZObject, bundle: Bundle): Promise<any[]> {
  const {documentId, tableId} = bundle.inputData;
  const response = await listColumns({
    z,
    bundle,
    documentId,
    tableId,
    pageToken: bundle.meta.page ? await z.cursor.get() : undefined,
  });
  if (response.nextPageToken) {
    await z.cursor.set(response.nextPageToken);
  }
  return response.items;
}

export const ColumnResource = {
  key: 'column' as 'column',
  noun: 'Column',

  list: {
    display: {
      label: 'New Column',
      description: 'Triggers when a new column is added to the selected table.',
      hidden: true,
    },
    operation: {
      canPaginate: true,
      inputFields: [
        {
          key: 'documentId',
          required: true,
          label: 'Document',
          dynamic: 'documentList.id.name',
          altersDynamicFields: true,
        },
        {
          key: 'tableId',
          required: true,
          label: 'Table',
          dynamic: 'tableList.id.name',
          altersDynamicFields: true,
          helpText: 'Changes to your document may take a little time to show up here.',
        },
      ],
      perform: performListColumns,
      sample: {
        items: [
          {
            id: 'c-abc123',
            name: 'Column A',
            href: 'https://coda.io/apis/v1/docs/12345/tables/grid-abc123/columns/c-abc123',
          },
          {
            id: 'c-def456',
            name: 'Column B',
            href: 'https://coda.io/apis/v1/docs/12345/tables/grid-abc123/columns/c-def456',
          },
        ],
      },
    },
  },

  sample: {
    id: 'c-102sdfn28',
    name: 'Column A',
    href: 'https://coda.io/apis/v1/docs/12345/tables/grid-abc123/columns/c-102sdfn28',
  },

  outputFields: [
    {key: 'id', label: 'ID'},
    {key: 'name', label: 'Name'},
    {key: 'href', label: 'Link'},
    {key: 'calculated', label: 'Calculated'},
    {key: 'display', label: 'Display'},
  ],
};
```

**Comment by xavdid** [comment]
nice! the `appTester(App.resources.column.list.operation.perform, bundle);` is the key here - that's the correct test code, but it's running afoul of the lines caleb linked in the OP. I'll get this fixed!

**Comment by codajonathan** [comment]
Very much appreciate the quick response time here, thanks!

Relatedly, once fixed, is there a way to mock/hack setting `z.cursor` in the test? It's obviously easy to pass in an arbitrary bundle, but to meaningfully test cursor-based pagination one would need to set the cursor to something non-empty. But that seems like it's all within the bowels of the app tester.

**Comment by xavdid** [comment]
Yeah, great question! once the fix is deployed (which, pre-warning, can take some time) and you've got the update, then your tests with cursors will use the actual cursor endpoints and perform just like production code will. The apptester generates (short-lived) cursors for each call to `appTester(...` (so each test gets its own cursor)

If you (understandably) want to fix your tests right now you can edit this function in your local `node_modules` to be `return true`: https://github.com/zapier/zapier-platform/blob/658171f97487f2df30d26f642e8377201b284c88/packages/core/src/tools/create-app-tester.js#L13-L18

That'll mean that all your tests will be able to use triggers (even when they shouldn't), but that's the easiest guaranteed fix. That'll keep working until you re-install `node_modules`, but it'll at least unblock you.

**Comment by xavdid** [comment]
This has been released as part of CLI `10.0.1`.
